game-app/game-core/src/main/java/games/strategy/triplea/ui/ActionButtons.java

```diff
@@ -255,27 +278,40 @@ public UserActionAttachment waitForUserActionAction(
    * @return null if no placement was made.
    */
   public PlaceData waitForPlace(final boolean bid, final IPlayerBridge bridge) {
+    if (placePanel == null) {
+      return null;
+    }
     return placePanel.waitForPlace(bid, bridge);
   }
 
   /** Blocks until the user selects an end-of-turn action. */
   public void waitForEndTurn(final TripleAFrame frame, final IPlayerBridge bridge) {
-    endTurnPanel.waitForDone(frame, bridge);
+    if (endTurnPanel != null) {
+      endTurnPanel.waitForDone(frame, bridge);
+    }
   }
 
   public void waitForMoveForumPosterPanel(final TripleAFrame frame, final IPlayerBridge bridge) {
-    moveForumPosterPanel.waitForDone(frame, bridge);
+    if (moveForumPosterPanel != null) {
+      moveForumPosterPanel.waitForDone(frame, bridge);
+    }
   }
 
   /** Blocks until the user selects a battle to fight. */
   public FightBattleDetails waitForBattleSelection() {
+    if (battlePanel == null) {
+      return null;
+    }
     return battlePanel.waitForBattleSelection();
   }
 
   public Tuple<Territory, Set<Unit>> waitForPickTerritoryAndUnits(
       final List<Territory> territoryChoices,
       final List<Unit> unitChoices,
       final int unitsPerPick) {
+    if (pickTerritoryAndUnitsPanel == null) {
+      return Tuple.of(null, Set.of());
```

Comment:
Are the usages of this null safe? Is there a trade-off to just return simply `null`?